pkg/tests/end_to_end_tests/datasets_test.go

```diff
@@ -413,3 +420,56 @@ func generateRandomMetricMetadata(num int) []prompb.MetricMetadata {
 	}
 	return data
 }
+
+func readTraces(t testing.TB, count int) []pdata.Traces {
+	// Clamp to max.
+	if count > tracesEntryCount {
+		count = tracesEntryCount
+	}
+
+	f, err := os.Open("../testdata/traces-dataset.sz")
```

Comment:
I think it would be good to document at a high-level how was this generated. Was the file output by the opentelemetry collector then compressed? what opentelemetry exporter did you use? Did you use a special program? etc.